src/imu/imu.cpp

```diff
@@ -53,6 +53,14 @@ void setupICM(void)
 bool getAccelerometer(float *acc_x, float *acc_y, float *acc_z)
 {
 #if SF_PLATFORM == SF_PLATFORM_PARTICLE
+
+    myICM.getAGMT();
+
+    if (myICM.status != ICM_20948_Stat_Ok)
+    {
+        FLOG_AddError(FLOG_ICM_FAIL, 0);
+        return false;
+    }
     ICM_20948_AGMT_t agmt = myICM.getAGMT();
```

Comment:
Lines 57 and 64 appear to have duplicate behavior.  Would it be simpler to move line 64 earlier?